lib/models/blueprint.js

```diff
@@ -770,7 +864,45 @@ let Blueprint = CoreObject.extend({
 
     this._ignoreUpdateFiles();
 
-    return finishProcessingForUninstall(fileInfos.filter(isValidFile));
+    fileInfos = fileInfos.filter(isValidFile).reduce((acc, info) => {
```

Comment:
For the polyfill, you can move htis logic into `_getFileInfos` directly (you will have to test that you are in the uninstall pathway before doing the new logic, but you should be able to determine that fairly easily)